src/cli/commands/set_api_endpoint_test.go

```diff
@@ -0,0 +1,187 @@
+package commands_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	. "cli/commands"
+	"cli/ui"
+
+	"code.cloudfoundry.org/cli/cf/configuration/confighelpers"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/ghttp"
+)
+
+var _ = Describe("API Endpoint Cmd", func() {
+
+	var (
+		command        *ApiCommand
+		configFilePath string
+		content        []byte
+		err            error
+	)
+
+	BeforeEach(func() {
+		defaultCFConfigPath, _ := confighelpers.DefaultFilePath()
+		targetsPath := filepath.Join(filepath.Dir(defaultCFConfigPath), "plugins", "AutoScalerConfig")
+		configFilePath = filepath.Join(targetsPath, "config.json")
+
+	})
+
+	Context("Set API endpoint", func() {
+
+		Context("with http server", func() {
+			var apiServer *ghttp.Server
+
+			BeforeEach(func() {
+				apiServer = ghttp.NewServer()
+				apiServer.RouteToHandler("GET", "/health",
+					ghttp.RespondWithJSONEncoded(http.StatusOK, ""),
+				)
+
+				command = &ApiCommand{
+					OptionalArgs: APIPositionalArgs{
+						URL: apiServer.URL(),
+					},
+					Unset:             false,
+					SkipSSLValidation: false,
+				}
+			})
+
+			AfterEach(func() {
+				apiServer.Close()
+			})
+
+			It("Return a valid json in config file", func() {
+				err = command.Execute([]string{})
```

Comment:
again, try to move the action to BeforeEach or JustBeforeEach, and in It() do the assertion